### PR TITLE
Feat v4 floatbutton

### PR DESCRIPTION
**Pull request by shifeng1993**
First of all, thank you for your contribution! 😄

New feature please send pull request to feature branch, and rest to master branch. Pull request will be merged after one of collaborators approve. Please makes sure that these form are filled before submitting your pull request, thank you!

[[中文版模板 / Chinese template](https://github.com/vueComponent/ant-design-vue/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / document update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Branch merge
- [ ] Other (about what?)

### What's the background?

> 1. Describe the source of requirement.
> 2. Resolve what problem.
> 3. Related issue link.

### API Realization (Optional if not new feature)

> 1. Basic thought of solution and other optional proposal.
> 2. List final API realization and usage sample.
> 3. GIF or snapshot should be provided if includes UI/interactive modification.

### What's the effect? (Optional if not new feature)

> 1. Does this PR affect user? Which part will be affected?
> 2. What will say in changelog?
> 3. Does this PR contains potential break change or other risk?

### Changelog description (Optional if not new feature)

> 1. English description
> 2. Chinese description (optional)

### Self Check before Merge

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### Additional Plan? (Optional if not new feature)

> If this PR related with other PR or following info. You can type here.

### CODE REVIEW

File: site/index.html [hunk 1]
```diff
@@ -81,7 +81,7 @@
       data-sf-width="350px"
       data-sf-height="450px"
       data-sf-preload="true"
-    ></div>
+    ></div> -->
```
Comment (shifeng1993) [inline site/index.html:84]: 广告把 floatbutton盖住了，暂时注释掉

**Comment by shifeng1993** [comment]
以前的 backtop 文件夹要删掉吗？

**Comment by tangjinzhou** [comment]
> 以前的 backtop 文件夹要删掉吗？

先不删，review 的时候我再看看

File: components/float-button/FloatButton.tsx [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+
+    const mergeShape = computed(() => {
+      return groupShape || props.shape;
+    });
+
+    const classString = computed(() => {
+      return classNames(
+        hashId.value,
+        prefixCls.value,
+        attrs.class,
```
Comment (tangjinzhou) [inline components/float-button/FloatButton.tsx:38]: attrs.class, 不是响应式的

File: components/float-button/FloatButton.tsx [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+            {buttonNode}
+          </button>
+        ),
+      );
+    };
+  },
+});
+
+if (process.env.NODE_ENV !== 'production') {
+  FloatButton.displayName = 'FloatButton';
```
Comment (tangjinzhou) [inline components/float-button/FloatButton.tsx:124]: 同上一个组件

File: components/float-button/FloatButtonContent.tsx [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+import FileTextOutlined from '@ant-design/icons-vue/FileTextOutlined';
+import classNames from '../_util/classNames';
+import { initDefaultProps } from '../_util/props-util';
+import { floatButtonContentProps } from './interface';
+
+const FloatButtonContent = defineComponent({
+  compatConfig: { MODE: 3 },
+  name: 'AFloatButtonContent',
+  inheritAttrs: false,
+  props: initDefaultProps(floatButtonContentProps(), {}),
```
Comment (tangjinzhou) [inline components/float-button/FloatButtonContent.tsx:11]: 没有默认值就不用加了

**Review by tangjinzhou — CHANGES_REQUESTED** [review]

File: site/src/router/index.js [hunk 5]
```diff
@@ -38,6 +38,23 @@ const routes = [
         },
         component: () => import('../../../components/layout/demo/index.vue'),
       },
+      {
```
[the remark this answers is not in the document]
Comment (shifeng1993) [reply site/src/router/index.js:41]: 这里放的是iframe套的例子，我试了一下重启服务 路由构建脚本 没有添加上

**Comment by shifeng1993** [comment]
这次提交的 文件 components/_util/transition.tsx
会影响到floatbutton  groupmenu的动画样式 

https://github.com/vueComponent/ant-design-vue/commit/f39d4894e4b5a7646574658a6ce97a1d560c96b9

components/float-button-cn#components-float-button-demo-group-menu

**Comment by tangjinzhou** [comment]
> 这次提交的 文件 components/_util/transition.tsx 会影响到floatbutton groupmenu的动画样式
> 
> [f39d489](https://github.com/vueComponent/ant-design-vue/commit/f39d4894e4b5a7646574658a6ce97a1d560c96b9)
> 
> components/float-button-cn#components-float-button-demo-group-menu

这个我来看看

有兴趣的话留个微信号，拉你进我们微信群

File: components/float-button/FloatButtonGroup.tsx [hunk 6]
```diff
@@ -33,7 +33,7 @@ const FloatButtonGroup = defineComponent({
     const floatButtonRef = ref<HTMLButtonElement | HTMLAnchorElement>(null);
 
     FloatButtonGroupContext.useProvide({
-      shape: computed(() => props.shape),
+      shape: props.shape,
```
Comment (tangjinzhou) [inline components/float-button/FloatButtonGroup.tsx:36]: 改成这样就丢失响应式了吧

**Review by tangjinzhou — CHANGES_REQUESTED** [review]

**Comment by shifeng1993** [comment]
> > 这次提交的文件components/_util/transition.tsx会影响到floatbutton groupmenu的动画样式
> > [f39d489](https://github.com/vueComponent/ant-design-vue/commit/f39d4894e4b5a7646574658a6ce97a1d560c96b9)
> > 组件/float-button-cn#components-float-button-demo-group-menu
> 
> 这个我来看看
> 
> 有兴趣的话留个微信，拉你进入我们微信群@shifeng1993

sf542674656

**Comment by shifeng1993** [comment]
createContext  这个函数用 reactive包一层  内部的ref，在用ref的时候会去掉.value 很奇怪。

**Comment by tangjinzhou** [comment]
> 我看看

**Pull request merged**